### ISSUE TITLE
No 'sensor' entity when casting to Google Next Hub v2

### ISSUE THREAD

**Issue by michaelblight** [thomasloven/hass-browser_mod#333]
My Home Assistant version: 2022.6.1

**What I am doing:**
I'm casting to a Google Home display. The instructions say to include resources in `ui_lovelace.yaml`, which I have done, despite
this method being outdated - you get DEPRECATED messages in VS-code, and similarly if I put it in the source of the page I get the following:
```
Resources should no longer be added to the dashboard configuration but can be added in the dashboard config panel.
```
Consequently I have also added `/browser_mod.js` as a module into the dashboard config panel. So I have it in all 3 places (`ui_lovelace.yaml`; page source; dashboard config). If I try a different path in the dashboard config (eg. `/custom_components/browser_mod/browser_mod.js`) and restart, none of the entities change from "unavailable". This gives me confidence that `/browser_mod.js` in the dashboard config is actually doing something.

**What I expected to happen:**
I expect to see a `sensor.*` entity created. I do get them created with normal browsers (eg. Chrome on Windows and iPad).

**What happened instead:**
I get the light, number, and switch entities, but not the sensor.

**Minimal** steps to reproduce:

```yaml
# The least amount of code or steps possible to reproduce my error

# Put your code/steps here
```

**Error messages from the browser console:**

```
// Select everything from the browser console and copy it

// Paste it here
```

---

**By replacing the space in the checkboxes ([ ]) with an X below, I indicate that I:**

- [X] Understand that this is a channel for reporting bugs, not a support forum (https://community.home-assistant.io/).

- [X] Have made sure I am using the latest version of the plugin.

- [X] Have followed the troubleshooting steps of the "Common Problems" section of https://github.com/thomasloven/hass-config/wiki/Lovelace-Plugins.

- [X] Understand that leaving one or more boxes unticked or failure to follow the template above may increase the time required to handle my bug-report, or cause it to be closed without further action.

**Comment by Andrewpaps** [comment]
Same issue occurring. No matter where the resource is added, browsermod features don't work (popups etc.)

**Comment by swiergot** [comment]
Try this in configuration.yaml:

```
browser_mod:

lovelace:
  mode: yaml
  resources:
    - url: /browser_mod.js
      type: module
```

If you install browser_mod from HACS, you don't need to care about file locations.

**Comment by Andrewpaps** [comment]
> 

This method doesn't seem to work. The initial install was through HACS. Putting it into yaml mode also unloads all of my other HACS addons.

**Comment by swiergot** [comment]
Well it works for me. Perhaps something completely different is going on since you get all other entities, i.e. it's not a matter of loading the resource.

**Comment by Andrewpaps** [comment]
When I try to load in a browser player card to see if it can find a device id, I get the error message 'Custom element does not exist: browser player. But it works everywhere else.

For reference, a sensor is created on my Google Nest Hub but not for my Google Nest Hub Max.

**Comment by Andrewpaps** [comment]
Also, calling the device specifically with it's deviceID does nothing, but replacing it with others works on those devices.

service: browser_mod.popup
data:
  title: PC Power Options
  deviceID:
    - af1145a3-803895cf
  large: true
  hide_header: false
  auto_close: false
  time: 20
  card:
    type: custom:select-list-card
    entity: input_select.pc_power_options
    title: ''
    icon: mdi:television-box
    show_toggle: false
    truncate: true
    scroll_to_selected: true
    max_options: 10

**Comment by abigdeel** [comment]
I'm having this same problem. There's a closed issue as well #288 that seems to be the same thing.

Nest Hub v2, browser_mod doesn't load when using Home Assistant Cast. No popups or the browser_player show anything. 
Is this a casting issue? Also the first "hub" device I've gotten. They seem pretty limited.

Running 2022.8.6 and up to date.

**Comment by saucedge** [comment]
Also having the same problem with my nest hub v2.  I have a nest hub v1 that works fine.  But, the v2 hub generates the media_player and light entity WITHOUT the sensor entity.

**Comment by abigdeel** [comment]
> Also having the same problem with my nest hub v2. I have a nest hub v1 that works fine. But, the v2 hub generates the media_player and light entity WITHOUT the sensor entity.

Good to know. 

There's specific mention in the browser_mod Github install instructions for adding "resources:" so it works with Home Assistant Cast. But that doesn't seem to work for the Nest Hub v2 and maybe the Nest Hub Max.

I've tried all combinations of adding the resources as well, manually in the YAML and through the Dashboard Resources page. No luck, and also no guaranteed way of checking either since you can't clear the cache on these devices. But I did restart the Hub a bunch while trying.

**Comment by saucedge** [comment]
I'm exclusively using CATT in my attempts.  I've also tried adding the Browser_Mod resources in various ways with no change in behavior.

**Comment by swiergot** [comment]
Ok. I have Lenovo Smart Display 7 and all is working fine, so probably it's closer to Nest Hub v1.

**Comment by abigdeel** [comment]
I haven't used CATT. Only just got the Hub and it seemed like HA Cast was the easier method.

There's also styling missing, so card-mod doesn't seem to be working either.

**Comment by thomasloven** [comment]
Closed because of Browser Mod 2.0 release.
Please see #354.
